### PR TITLE
Creating unit tests in protocols

### PR DESCRIPTION
**Pull request by spetebroodt**

### CODE REVIEW

File: develop/best_practices/Best_practices.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+uid: Best_practices
+---
+
+# Best practices
+
+In this section:
+
+- [Creating unit tests in protocols](xref:Creating_unit_tests_in_protocols)
```
Comment (PedroDebevere) [inline develop/best_practices/Best_practices.md:9]:
```suggestion
- [Creating unit tests](xref:Creating_unit_tests_in_protocols)
```
I would keep the title more general "Creating unit tests" as the explanation is also applicable to e.g. Automation scripts or solutions in general (the text does explain it for a protocol but it's very similar for Automation scripts).

File: develop/best_practices/Creating_unit_tests_in_protocols.md [hunk 2]
```diff
@@ -0,0 +1,638 @@
+---
+uid: Creating_unit_tests_in_protocols
```
Comment (PedroDebevere) [inline develop/best_practices/Creating_unit_tests_in_protocols.md:2]: This page is quite big, maybe it's better to make subpages corresponding to the blog post series: https://community.dataminer.services/unit-testing-data-driven-tests/
I think the article for Data-driven tests is also missing here.

File: develop/index.md [hunk 3]
```diff
@@ -14,3 +14,4 @@ uid: DevelopIndex
 - <xref:DSI>
 - [DataMiner module integrations](xref:DataMinerModuleIntegrations)
 - <xref:ThirdPartyIntegrations>
+- [Best practices](xref:Best_practices)
```
Comment (PedroDebevere) [inline develop/index.md:17]: an entry for the newly added pages will also need to be added to the develop/toc.yml

Comment (spetebroodt) [reply develop/index.md:17]: will do afterwards. had a merge conflict before.

**Comment by spetebroodt** [comment]
Hi @PedroDebevere , can you have another look at this PR? I now added the rest of the unit testing blog posts.
Note that I haven't added the best practices to the main TOC yet in order to avoid merge conflicts. I'll add this at the last moment before merging.

File: develop/best_practices/Creating_unit_tests_in_protocols.md (hunk 2 above)
Comment (spetebroodt) [reply develop/best_practices/Creating_unit_tests_in_protocols.md]: Done!

**Pull request merged**